spec/requests/organizations/service_agreements_controller_spec.rb

```diff
@@ -132,4 +132,267 @@
+              email: '-1'
+            }
+          }
+        }
+      }
+    end
+
+    context 'when a organization administrator is signed in' do
+      let(:organization_administrator) { create :organization_member, organization: organization }
+
+      before do
+        sign_in organization_administrator.user
+        perform_request
+      end
+
+      context 'when there is no civi set' do
+        it 'successfully renders the new service agreement form' do
+          expect(response).to render_template(:new)
+        end
+      end
+
+      context 'when there is a civi set' do
+        let(:request_params) do
+          {
+            service_agreement: {
+              civil_servant_attributes: {
+                user_attributes: {
+                  email: maria.user.email
+                }
+              }
+            }
+          }
+        end
+
+        it 'successfully renders the new service agreement form' do
+          expect(response).to render_template(:new)
+        end
+      end
+    end
+
+    context 'when a civil servant is signed in' do
+      let(:civil_servant) { create :civil_servant, :full }
+
+      before { sign_in civil_servant.user }
+
+      it_behaves_like 'unauthorized request' do
+        before { perform_request }
+      end
+    end
+
+    context 'when nobody is signed in' do
+      before { perform_request }
+
+      it_behaves_like 'unauthenticated request'
+    end
+  end
+
+  describe '#create' do
+    let(:perform_request) { post(organizations_service_agreements_path, params: params) }
+    let(:params) { { service_agreement: service_agreement_params } }
+
+    let(:organization) { create :organization }
+    let(:service_specification) { create :service_specification, organization: organization }
+
+    let(:valid_service_agreement_params) do
+      attributes_for(:service, :civil_servant_agreement_pending, :unconfirmed, :future)
+        .slice(:beginning, :ending, :service_type)
+        .merge(service_specification_id: service_specification.id)
+    end
+
+    context 'when a organization administrator is signed in' do
+      let(:organization_administrator) { create :organization_member, organization: organization }
+
+      before do
+        sign_in organization_administrator.user
+      end
+
+      context 'with existing civil servant' do
+        let!(:civil_servant) { create(:civil_servant, :full) }
+        let(:service_agreement_params) do
+          valid_service_agreement_params.merge(
+            civil_servant_attributes: {
+              user_attributes: {
+                email: civil_servant.user.email
+              }
+            }
+          )
+        end
+
+        let(:created_service_agreement) { Service.order(:created_at).last }
+
+        context 'with valid parameters' do
+          it 'creates a new service agreement and redirects back to the service specifications list' do
+            expect { perform_request }.to change(Service, :count).by(1)
```

Comment:
Can you also add some expectations to the created service? E.g. that it is actually linked with the passed civil servant?